src/platform/update/new_release_monitor.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_NEW_RELEASE_MONITOR_H
+#define MULTIPASS_NEW_RELEASE_MONITOR_H
+
+#include <multipass/qt_delete_later_unique_ptr.h>
+#include <multipass/optional.h>
+#include <multipass/new_release_info.h>
+#include "version.h"
+
+#include <QObject>
+#include <QString>
+#include <QTimer>
+
+namespace multipass
+{
+class LatestReleaseChecker;
+
+/*
+ * NewReleaseMonitor - monitors for new Multipass releases
+ *
+ * Checks for new releases occur on an internal thread to avoid blocking
+ * callers. Rechecks happen at a specified intervals.
+ *
+ * TODO: the internal thread can be removed when refactoring to move all
+ * blocking calls off the main-loop.
+ */
+
+class NewReleaseMonitor : public QObject
+{
+    Q_OBJECT
+public:
+    NewReleaseMonitor(const QString& current_version, std::chrono::hours refresh_rate);
```

Comment:
Is there a more generic type `refresh_rate` could be?

Reply:
I think std::chrono::duration would do. The github api limits you to 30 checks per day, so I thought this would be granular enough. Happy to change

Reply:
I'm more thinking we could do less often. Just felt weird to impose `hour`s here.

Reply:
Am using more generic type now